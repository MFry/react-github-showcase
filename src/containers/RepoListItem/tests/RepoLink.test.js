import React from 'react';
import { mount, shallow } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import 'jest-styled-components';

import RepoLink from '../RepoLink';

describe('<RepoLink />', () => {
  it('should match the snapshot', () => {
    expect(shallow(<RepoLink />)).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const wrapper = mount(<RepoLink />);
    const renderedComponent = enzymeFind(wrapper, RepoLink);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(<RepoLink id={id} />);
    const renderedComponent = enzymeFind(wrapper, RepoLink);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(<RepoLink attribute="test" />);
    const renderedComponent = enzymeFind(wrapper, RepoLink);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
