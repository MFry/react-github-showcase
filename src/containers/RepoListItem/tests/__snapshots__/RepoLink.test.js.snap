// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoLink /> should match the snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  color: #41addd;
  &:hover {
    color: #6cc0e5;
  }
",
          "
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  width: 100%;
",
        ],
      },
      "displayName": "Styled(styled.a)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
